ext/RMagick/rmimage.c

```diff
@@ -13792,15 +13792,15 @@ Image_transparent(int argc, VALUE *argv, VALUE self)
 {
     Image *image, *new_image;
     MagickPixel color;
-    Quantum opacity = TransparentOpacity;
+    Quantum alpha = TransparentAlpha;
     MagickBooleanType okay;
 
     image = rm_check_destroyed(self);
 
     switch (argc)
     {
         case 2:
-            opacity = APP2QUANTUM(argv[1]);
+            alpha = rm_get_named_alpha_value(argv[1], "Image#transparent", "alpha");
```

Comment:
if `rm_get_named_alpha_value` takes only constant name `"alpha"`, it is better to pass pre-generated Symbol object.

```
void
Init_RMagick2(void)
{
    ...
    sym_alpha = ID2SYM(rb_intern(argument_name));
    ...
}
```

```
    alpha = rm_get_named_alpha_value(argv[1], "Image#transparent", sym_alpha);
```


like https://github.com/brianmario/mysql2/blob/4e221cfdec4edfc08dfd1bf0ee405d695af813ee/ext/mysql2/result.c#L701

Reply:
The name of the argument can be different. `Image#tint` has the names `red_alpha`, `green_alpha`, etc.. We need to create `sym_[name]` values for all of them if we want this cached?

Reply:
OK, we can cache them later if necessary